plugins/guests/haiku/cap/configure_networks.rb

```diff
@@ -0,0 +1,29 @@
+require "tempfile"
+
+require_relative "../../../../lib/vagrant/util/template_renderer"
+
+module VagrantPlugins
+  module GuestHaiku
+    module Cap
+      class ConfigureNetworks
+        include Vagrant::Util
+
+        def self.configure_networks(machine, networks)
+          networks.each do |network|
+            entry = TemplateRenderer.render("guests/haiku/network_#{network[:type]}",
+                                            options: network)
+            Tempfile.open("vagrant-haiku-configure-networks") do |f|
+              f.binmode
+              f.write(entry)
+              f.fsync
+              f.close
+              machine.communicate.upload(f.path, "/tmp/vagrant-network-entry")
+            end
+            machine.communicate("mv /tmp/vagrant-network-entry /boot/system/settings/network/interfaces")
```

Comment:
```suggestion
            machine.communicate.execute("mv /tmp/vagrant-network-entry /boot/system/settings/network/interfaces")
```

Should there be multiple interfaces in `/boot/system/settings/network/interfaces`? With multiple networks defined, I ended up with only a single interface:

Vagrantfile:

```ruby
  config.vm.network "private_network", type: "dhcp"
  config.vm.network "public_network", ip: "192.168.127.2"
```

/boot/system/settings/network/interfaces:

```
interface  {
  address inet {
    address 192.168.127.2
    mask 255.255.255.0
  }
}
```

I think you'll want to move the `Tempfile` creation outside of the loop so we keep writing to the same interfaces file.

Reply:
Ah. good catch. yeah, that'll only ever template out the first interface.